source/MaterialXShaderGen/HwShaderGenerator.cpp

```diff
@@ -21,7 +21,7 @@ void HwShaderGenerator::bindLightShader(const NodeDef& nodeDef, size_t lightType
 
     if (getBoundLightShader(lightTypeId))
     {
-        throw ExceptionShaderGenError("Error binding light shader. Light type id '" + std::to_string(lightTypeId) + "' has already been bound");
+        return;
```

Comment:
This exception is here to make sure you don't register the same light type id again, but with a different shader. If this happens now it will silently accept this and your light shader will be a different one that you might think. But also, registering the same light type id twice even if the shader is the same is unnecessary and I think we should throw here  to make users aware when that happens.